### ISSUE TITLE
When I close the shiny's app window while the app is running R session aborted.

### ISSUE THREAD

**Issue by MihalisPapatsimpas** [rstudio/shiny#2035]

**Comment by MihalisPapatsimpas** [comment]
When i press the actionbutton apply the code starts running.
![22](https://user-images.githubusercontent.com/17166680/39098933-833cdbdc-467a-11e8-8745-99f60bcf3093.png)

But if i go and close the app window while the app is running i get this.
![24](https://user-images.githubusercontent.com/17166680/39098944-ad5c9916-467a-11e8-923c-361419df4ef9.png)

Is there any way to fix it?

**Comment by MihalisPapatsimpas** [comment]
─ Session info ────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.4.4 (2018-03-15)
 os       Windows >= 8 x64            
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  Greek_Greece.1253           
 tz       Europe/Istanbul             
 date     2018-04-22                  

─ Packages ────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version date       source        
 clisymbols    1.2.0   2017-05-21 CRAN (R 3.4.4)
 digest        0.6.15  2018-01-28 CRAN (R 3.4.3)
 htmltools     0.3.6   2017-04-28 CRAN (R 3.4.3)
 httpuv        1.4.0   2018-04-19 CRAN (R 3.4.4)
 later         0.7.1   2018-03-07 CRAN (R 3.4.4)
 magrittr      1.5     2014-11-22 CRAN (R 3.4.4)
 mime          0.5     2016-07-07 CRAN (R 3.4.1)
 promises      1.0.1   2018-04-13 CRAN (R 3.4.4)
 R6            2.2.2   2017-06-17 CRAN (R 3.4.3)
 Rcpp          0.12.16 2018-03-13 CRAN (R 3.4.4)
 sessioninfo   1.0.0   2017-06-21 CRAN (R 3.4.4)
 shiny       * 1.0.5   2017-08-23 CRAN (R 3.4.4)
 withr         2.1.2   2018-03-15 CRAN (R 3.4.4)
 xtable        1.8-2   2016-02-05 CRAN (R 3.4.3)

**Comment by wch** [comment]
This could be due to a problem in upgrading packages.

Try the following:

Restart a clean R session (and quit any other ones), and then reinstall httpuv, then reinstall shiny.

If you still have problems after that, please include a minimal reproducible example app.

**Comment by MihalisPapatsimpas** [comment]
`
library(shiny)
library(quantmod)
ui <- fluidPage(
   titlePanel("Data"),
   sidebarLayout(
      sidebarPanel(
        actionButton("h","hit")
      ),
      mainPanel(
         plotOutput("plot")
      )
   )
)
getRate <- function(name){
  pair<-getSymbols(name,src = "oanda",from = Sys.Date()-179,to =Sys.Date(),auto.assign = FALSE)
  return(pair)
}
compute_pips <- function(name){
  pair <- getRate(name)
  cur<-as.matrix(pair)
  pips<-cur
  for(i in 1:length(pips)-1){
    pips[i]<-10000*(cur[i+1]-cur[i])
  }
  return(pips)
}
data <- "EUR/USD"
server <- function(input, output) {
   observeEvent(input$h,{
     pips<-compute_pips(data)
     output$plot <- renderPlot({
        chartSeries(pips)
     })
   })

}
 
shinyApp(ui = ui, server = server)

`
`

**Comment by MihalisPapatsimpas** [comment]
@wch i reinstalled shiny from github (rstudio/shiny) ,and now the above app runs normal even if i close app window. But in my main app when i close the window after pressing the actionButton the debugger activates and shows me this code:
`function (con, bigint_as_char) {`
` `  `stopifnot(inherits(con, "connection"))`
` `   `if (!isOpen(con)) {`
` ` ` ` `on.exit(close(con))`
` ` ` `  `open(con, "rb")`
` ` ` ` `}`
` ` ` ` `.Call(R_parse_connection, con, bigint_as_char)`
`}`

And when i press continue or stop the app terminates normal.
What is the meaning of this and how do I solve it?

Thanks in advance!

**Comment by wch** [comment]
That error is coming from the jsonlite package: https://github.com/jeroen/jsonlite/blob/ae0ef71d/R/push_parser.R

As of yet, I don't know why that's happening.

I'm not able to reproduce the problem when I run your code. (Note: the line with `data <-` needed to be fixed.)

Did you reinstall httpuv as well? Please make sure that you do that in a new R session, and that no other R sessions are running. If that doesn't fix it, I suggest reinstalling all the packages. The cause of your problem may be similar to #2030.

Please also provide the full output of `sessioninfo::session_info()` -- it helps to know what platform you are running on but the previous output that you provided did not include that information.

**Comment by MihalisPapatsimpas** [comment]
Yes i reinstalled it again!
─ Session info ────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.4.4 (2018-03-15)
 os       Windows >= 8 x64            
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  Greek_Greece.1253           
 tz       Europe/Istanbul             
 date     2018-04-23                  

─ Packages ────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version date       source        
 clisymbols    1.2.0   2017-05-21 CRAN (R 3.4.4)
 sessioninfo   1.0.0   2017-06-21 CRAN (R 3.4.4)
 withr         2.1.2   2018-03-15 CRAN (R 3.4.4)

**Comment by wch** [comment]
Can you try running the following and provide the output from it?

```R
tools::checkMD5sums("httpuv")
tools::checkMD5sums("htmltools")
```

**Comment by MihalisPapatsimpas** [comment]
`tools::checkMD5sums("httpuv")`
[1] TRUE
`tools::checkMD5sums("htmltools")`
[1] TRUE

**Comment by MihalisPapatsimpas** [comment]
@wch I made an another simpler app and still I get an error when i press the 'X' button to close the app window. So the logic is clear,when the actionButton is pressed print 1000 times "hi". So if you just close the app window before you press the button or after the loop has completed it closes fine. But if you try to close the window while it is printing it crashes. I tried with tryCatch to avoid the crash but it didn't handle the error. Here is the app:

`ui <- fluidPage(`
` ` `titlePanel("Data"),`
` ` `sidebarLayout(`
` ` ` ` `sidebarPanel(`
` ` ` ` ` ` `actionButton("h","hit")`
` ` ` ` ` ),`
` ` ` ` `mainPanel()`
` ` ` )`
` )`

`server <- function(input, output) {`
` ` `observeEvent(input$h,{`
` ` ` ` ` for(i in 1:1000){`
` ` ` ` ` ` `print("hi")`
` ` ` ` ` }`
` ` ` })`
`}`
`shinyApp(ui = ui, server = server)`
